homeassistant/util/json.py

```diff
@@ -61,17 +62,16 @@ def save_json(
         raise SerializationError(msg) from error
 
     tmp_filename = ""
-    tmp_path = os.path.split(filename)[0]
     try:
         # Modern versions of Python tempfile create this file with mode 0o600
         with tempfile.NamedTemporaryFile(
-            mode="w", encoding="utf-8", dir=tmp_path, delete=False
+            mode="w", encoding="utf-8", delete=False
         ) as fdesc:
             fdesc.write(json_data)
             tmp_filename = fdesc.name
         if not private:
             os.chmod(tmp_filename, 0o644)
-        os.replace(tmp_filename, filename)
+        shutil.move(tmp_filename, filename)
```

Comment:
Why is this change needed?

Reply:
When using os.replace "OSError: [Errno 18] Invalid cross-device link" may be reported. shutils.move is safe against such case.